src/main/java/com/metamx/cache/CaffeineCache.java

```diff
@@ -78,7 +93,7 @@ public void put(NamedKey key, byte[] value)
   @Override
   public Map<NamedKey, byte[]> getBulk(Iterable<NamedKey> keys)
   {
-    return Maps.transformValues(cache.getAllPresent(keys), this::deserialize);
+    return ImmutableMap.copyOf(Maps.transformValues(cache.getAllPresent(keys), this::deserialize));
```

Comment:
It depends on how many times the value is expected to be used. There is a deserialization cost on each access.

The general assumption I'm making here is that each value will be accessed at least once (or else why pull it from cache?), and maybe more than once.